### DISCUSSION TITLE
Estimation for worker_pods_creation_batch_size for KubernetesExecutor

### DISCUSSION THREAD

**Discussion by lavsurgut** [apache/airflow#26472] [Q&A]
Dear Airflow Team,
`worker_pods_creation_batch_size` parameter is HIGHLY recommended to increase. Though the doc says ` ...to match the tolerance of their kubernetes cluster`. Could you please tell what might be a good value for it and how we can determine it. What kind of tolerance in k8s (e.g. likely it is kube-scheduler)?

**Comment by potiuk** [comment]
I think you shoudl experiment. There are quite a number of  parameters in Airlfow that depends on multiple factors - such as what kind of cluster you might have, whether it has some optimisations, how many nodes it has etc. I suggest you work out your own scheme of testing it to get the optimum performance. Where we can we can specifiy some recipes and rules-of-thumb, but here simply there are no such recipes, because there are too many factors that are impacting the limits you have.

For example (but this is just a suggestion) - increase it to 5, 10, 15, 20, .... and observe/measure if it performs better for your case and see if further increase will bring improvements - usually every such parameters has some moment when either you hit some limits of your environments, or simply law of diminishing returns kicks in and you won't get any more visible improvements in performance.

Fine tuning your deployment is really in your hands - in Airflow we give a lot of knobs to turn and if you need ot optimise the performance, it's in the hands of the deployment manager. Airflow is not self-tuning for all possible kind of cases and performance and you have to make a good deal of experimentation and tuning.

This is the best advice you can get for this. There is no way to tell "set it to 42 - this magic value is best" . I mean we know that 42 is the ultimate answer to live, universe and everything, but it might not work well in this case :)

**Reply by lavsurgut** [reply]
Haha. 42 might work in the end too!:)
I was wondering if it might be related to the kube-scheduler configuration. Apart from amount of nodes and nodes type, It seems that there is nothing else in k8s setup that could be affecting pod creation tolerance. Or anything else that might come to your mind? E.g.. i'm not really an expert in k8s and to me the Airflow doc sounds a little bit foggy. So, one can start worrying about it if you know that apart from Airflow you run some other critical workloads and you have no idea what it really means. In dev cluster one can experiment with it to some extent, but it is often by no means going to replicate proper setup that you have in prod (since it might be expensive to run full blown services in dev).

**Reply by potiuk** [reply]
No idea, really it might be but there might be more factors. I just - similarly as you - looked at the description of the parameter, but that's what I'd do if I needed to figure out the right value. I guess if there was a super straightforward relation, it would have been written there. 

However what I can advise you as well (if you think the doc is foggy) - this is a perfect opportunity to improve it. 
Why don't you experiment and find out what is the relation between the valua some limitations and describe it in the way that is less foggy? Airflow is an open-soure project that has > 2100 contributors - many of them like you who had some problems and needed some additional information/missed documentation, figured it out and contributed it back. It would be a great contribuition back for the free software you get.

**Reply by lavsurgut** [reply]
Sounds good. thanks
